client/hotrod-client/src/main/java/org/infinispan/client/hotrod/marshall/ProtoStreamMarshaller.java

```diff
@@ -5,24 +5,17 @@
 import org.infinispan.commons.marshall.Marshaller;
 import org.infinispan.protostream.ProtobufUtil;
 import org.infinispan.protostream.SerializationContext;
-import org.infinispan.query.remote.client.BaseProtoStreamMarshaller;
 
 /**
  * A client-side marshaller that uses Protocol Buffers.
  *
  * @author anistor@redhat.com
  * @since 6.0
  */
-public class ProtoStreamMarshaller extends BaseProtoStreamMarshaller {
-
-   private final SerializationContext serializationContext = ProtobufUtil.newSerializationContext();
+public class ProtoStreamMarshaller extends org.infinispan.commons.marshall.ProtoStreamMarshaller {
 
    public ProtoStreamMarshaller() {
-   }
-
-   @Override
-   public SerializationContext getSerializationContext() {
-      return serializationContext;
+      super(ProtobufUtil.newSerializationContext());
```

Comment:
There no-arg a super ctor already does ProtobufUtil.newSerializationContext()

Reply:
It seems this class continues to exist only for ```public static SerializationContext getSerializationContext(RemoteCacheManager remoteCacheManager)```. That could be a method in the RemoteCacheManager. That was not possible in the past because protobuf support had to be something entirely optional, hidden from all API. Do we still have that constraint?

Reply:
Do you expect this method to be used by users? We may need to keep it for compatibility reasons anyway, but if it's not for users we could just add it to `org.infinispan.client.hotrod.marshall.MarshallerUtil`

Reply:
It is used by users so far. We could think of preventing that in future, but for now that's how it works.

Reply:
> There no-arg a super ctor already does ProtobufUtil.newSerializationContext()

I see that constructor is sill there.